src/items.rs

```diff
@@ -2589,6 +2597,7 @@ fn rewrite_where_clause(
     terminator: &str,
     suppress_comma: bool,
     snuggle: bool,
+    compress_where: bool,
```

Comment:
I don't think you need to do it to land this PR, but I think there are enough bools here that it is worth factoring out a `WhereClauseOptions` struct.